cmds/core/dirname/dirname.go

```diff
@@ -9,17 +9,25 @@ package main
 
 import (
 	"fmt"
+	"io"
 	"log"
 	"os"
 	"path/filepath"
 )
 
-func main() {
-	if len(os.Args) < 2 {
-		log.Fatalf("dirname: missing operand")
+func run(args []string, stdout io.Writer) error {
```

Comment:
nitpick: `stdout` may not be the best name for this variable - the point of the `io.Writer` abstraction here is that we can write to stdout or a buffer or something else... so maybe the function argumetn should just be named `out`?

Reply:
fixed in 0aa81c8.